src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParameterNameHintsTag.cs

```diff
@@ -18,18 +33,97 @@ namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
-        public InlineParameterNameHintsTag(string text, double lineHeight, TextFormattingRunProperties format)
-            : base(CreateElement(text, lineHeight, format), removalCallback: null, PositionAffinity.Predecessor)
+        /// <param name="adornment">The adornment that we are creating</param>
+        /// <param name="textView">The view of the editor</param>
+        /// <param name="span">The span that has the location of the hint</param>
+        /// <param name="key">The symbolkey associated with each parameter</param>
+        private InlineParameterNameHintsTag(FrameworkElement adornment, ITextView textView, SnapshotSpan span,
+                                            SymbolKey key, InlineParameterNameHintsTaggerProvider taggerProvider)
+            : base(adornment, removalCallback: null, PositionAffinity.Predecessor)
+        {
+            _textView = textView;
+            _span = span;
+            _key = key;
+            _streamingPresenter = taggerProvider.StreamingFindUsagesPresenter;
+            _threadingContext = taggerProvider.ThreadingContext;
+            _toolTipService = taggerProvider.ToolTipService;
+
+            // Sets the tooltip to a string so that the tool tip opening event can be triggered
+            // Tooltip value does not matter at this point
+            adornment.ToolTip = "Quick info";
+            adornment.ToolTipOpening += Border_ToolTipOpening;
+        }
+
+        public static InlineParameterNameHintsTag Create(string text, double lineHeight, TextFormattingRunProperties format,
+                                                         ITextView textView, SnapshotSpan span, SymbolKey key,
+                                                         InlineParameterNameHintsTaggerProvider taggerProvider)
+        {
+            return new InlineParameterNameHintsTag(CreateElement(text, lineHeight, format), textView,
+                                                   span, key, taggerProvider);
+        }
+
+        public async Task<IReadOnlyCollection<object>> CreateDescriptionAsync(CancellationToken cancellationToken)
+        {
+            var document = _textView.TextBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var textContentBuilder = new List<TaggedText>();
+
+            if (document != null)
+            {
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = _key.Resolve(compilation, cancellationToken: cancellationToken).Symbol;
+                var workspace = document.Project.Solution.Workspace;
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var symbolDisplayService = workspace.Services.GetLanguageServices(semanticModel.Language).GetService<ISymbolDisplayService>();
+
+                if (symbolDisplayService != null)
```

Comment:
Can this happen? You've called GetRequiredSemanticModelAsync, so at that point we know it's only C# or VB and so they'll have to be a ISymbolDisplayService...check for the semantic model later?